src/hypothesis/searchstrategy/misc.py

```diff
@@ -77,7 +77,7 @@ class SampledFromStrategy(SearchStrategy):
 
     def __init__(self, elements):
         SearchStrategy.__init__(self)
-        self.elements = tuple(elements)
+        self.elements = d.check_sample(elements)
```

Comment:
Isn't this going to cause us to unconditionally raise the InvalidArgument error regardless of deprecation?